### PR TITLE
[Batch Submitter] Add MinL1TxSize and MaxL1GasPrice in batch submitter

### PR DESCRIPTION
**Pull request by boyuan-chen**
Add the MinL1TxSize  in the batch submitter, so it submits the batch if the batch size is larger than the `MinL1TxSize` or the waiting time is larger than `MaxBatchSubmissionTime`.
Add the MaxL1GasPrice in the batch submitter. It submits the batches only if the gas price is below `MaxL1GasPrice`.

### CODE REVIEW

**Comment by InoMurko** [comment]
Some tests would solidify the implementation and check for correctness.

File: ops/envs/batch-submitter.env [hunk 1]
```diff
@@ -3,9 +3,10 @@ ETH_NETWORK_NAME=clique
 
 LOG_LEVEL=debug
 BATCH_SUBMITTER_LOG_LEVEL=debug
+BATCH_SUBMITTER_MIN_L1_TX_SIZE=20000
 BATCH_SUBMITTER_MAX_L1_TX_SIZE=90000
-BATCH_SUBMITTER_MAX_BATCH_SUBMISSION_TIME=0
-BATCH_SUBMITTER_POLL_INTERVAL=500ms
+BATCH_SUBMITTER_MAX_BATCH_SUBMISSION_TIME=10s
```
Comment (boyuan-chen) [inline ops/envs/batch-submitter.env:8]: The batch submitter submits the batches if the batch size is larger than `MinL1TxSize` or the `timeSinceLastSubmission` is larger than 10s.

File: go/bss-core/drivers/interface.go [hunk 2]
```diff
@@ -41,4 +41,7 @@ type L1Client interface {
 	// TransactionReceipt returns the receipt of a transaction by transaction
 	// hash. Note that the receipt is not available for pending transactions.
 	TransactionReceipt(context.Context, common.Hash) (*types.Receipt, error)
+
+	// SuggestGasPrice returns the estimated gas price
+	SuggestGasPrice(context.Context) (*big.Int, error)
```
Comment (boyuan-chen) [inline go/bss-core/drivers/interface.go:46]: It was added for the testing purpose.

**Review by InoMurko — APPROVED** [review]
👏🏻 👏🏻 👏🏻

**Review by CAPtheorem — APPROVED** [review]

**Pull request merged**